config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
+  devise_for :users, controllers: { registrations: "registrations" }
+
+  resources :users
+
+  resources :budget_categories do
+    resources :expenses
+  end
```

Comment:
- [ ] I would recommend exposing only certain path which you want to show the users. You can learn more [from here](https://guides.rubyonrails.org/routing.html#configuring-the-rails-router).